org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -290,8 +285,10 @@ private PreparedStatement getOidStatement(String pgTypeName) throws SQLException
                 }
                 _getOidStatementSimple = _conn.prepareStatement(sql);
             }
+            // coerce to lower case to handle upper case type names
+            String lcName = pgTypeName.toLowerCase(Locale.ENGLISH);
```

Comment:
I actually don't think this is a good idea, but if you want such behavior probably a better way to go about it is to register built-in types and aliases as upercase.